### DISCUSSION TITLE
Is there a way to decluster jets in the awkward interface?

### DISCUSSION THREAD

**Discussion by sambt** [scikit-hep/fastjet#111] [Q&A]
Hi all,

I've been trying to use the fastjet python bindings to run soft-drop grooming on jets. I'm starting with a list of lists of particle kinematics that were clustered into AK8 jets (i.e. the constituents of the leading jet in each event), and can easily re-cluster these with Cambridge-Aachen very quickly using the awkward bindings:

```
p4s = ak.zip({"px":events.px,"py":events.py,"pz":events.pz,"E":events.e}) 
jetdef = fj.JetDefinition(fj.cambridge_algorithm,1.0)
cluster = fj.ClusterSequence(p4s,jetdef)
jets = cluster.inclusive_jets()
```

I want to traverse the C/A clustering history to run soft drop, but it seems there's no way to retrieve this. In principle, I would just need the order in which the individual particles were combined during C/A clustering. I'm currently falling back to the "normal" method of looping over the events and declustering/soft-dropping one jet at a time, but this is quite slow compared to the operations with awkward arrays.

**Comment by jpivarski** [comment]
For context, I recommended that @SamBT reposts his question here, coming from https://gitter.im/Scikit-HEP/community about 3 hours ago.

Since this function doesn't exist, this is a feature request. If he or someone else is willing to work on the implementation, I can provide how-to support.
